platform/ios/src/MGLMapViewDelegate.h

```diff
@@ -80,9 +136,67 @@ NS_ASSUME_NONNULL_BEGIN
     method returns `YES`, this camera becomes the map view’s camera.
  @return A Boolean value indicating whether the map view should stay at
     `oldCamera` or change to `newCamera`.
+
+ @note If `mapView:shouldChangeFromCamera:toCamera:reason:` is implemented this method will not be called.
  */
 - (BOOL)mapView:(MGLMapView *)mapView shouldChangeFromCamera:(MGLMapCamera *)oldCamera toCamera:(MGLMapCamera *)newCamera;
 
+/**
+ Asks the delegate whether the map view should be allowed to change from the
+ existing camera to the new camera in response to a user gesture.
+
+ This method is called as soon as the user gesture is recognized. It is not
+ called in response to a programmatic camera change, such as by setting the
+ `centerCoordinate` property or calling `-flyToCamera:completionHandler:`.
+
+ This method is called many times during gesturing, so you should avoid performing
+ complex or performance-intensive tasks in your implementation.
+
+ @param mapView The map view that the user is manipulating.
+ @param oldCamera The camera representing the viewpoint at the moment the
+ gesture is recognized. If this method returns `NO`, the map view’s camera
+ continues to be this camera.
+ @param newCamera The expected camera after the gesture completes. If this
+ method returns `YES`, this camera becomes the map view’s camera.
+ @param reason The reason for the camera change.
+ @return A Boolean value indicating whether the map view should stay at
+ `oldCamera` or change to `newCamera`.
+
+ @note If this method is implemented `mapView:shouldChangeFromCamera:toCamera:` will not be called.
+ */
+- (BOOL)mapView:(MGLMapView *)mapView shouldChangeFromCamera:(MGLMapCamera *)oldCamera toCamera:(MGLMapCamera *)newCamera reason:(MGLCameraChangeReason)reason;
+
+#pragma mark Responding to user gestures
+
+/**
+ Tells the delegate that the user has tapped on the map view.
+
+ This method will not be called when the user single taps on an annotation, or if an annotation
+ is currently selected.
+
+ @param mapView The map view that was tapped.
+ @param coordinate Location of tap in world coordinates.
+
+ @note If  you implement this method, custom tap gesture recognizers that are installed in the following
+ way
+
+```
+UITapGestureRecognizer *singleTap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleSingleTap:)];
+for (UIGestureRecognizer *gesture in self.mapView.gestureRecognizers)
+{
+    if ([gesture isKindOfClass:[UITapGestureRecognizer class]])
+    {
+        [singleTap requireGestureRecognizerToFail:gesture];
+    }
+}
+[self.mapView addGestureRecognizer:singleTap];
+```
+
+ will fail to trigger, since the built-in single tap recognizer will no longer fail.
+ */
+- (void)mapView:(MGLMapView *)mapView didSingleTapAtCoordinate:(CLLocationCoordinate2D)coordinate;
```

Comment:
Given the caveat above, I don’t think we should provide this method. I think it would be a slippery slope from this method to a number of gesture-specific methods. The tap gesture recognizer’s existing behavior of failing when tapping on empty space is elegant, takes advantage of system paradigms, and is attractive to developers. (After all, it was requested and implemented by contributors: #7246.)

This shortcut reminds me of `-mapView:didUpdateLocations:`, which has been a source of confusion for many developers; it tries to simplify an entry point for something that in reality everyone has slightly different expectations for.

Reply:
Maintaining the status quo here is also my preference.

Reply:
Leaving it in for my next push, but we should discuss. Wondering if long term this kind of functionality would be better broken into multiple protocols, perhaps even ripe for a plugin.

Reply:
A plugin would be appropriate if certain developers find the workaround described in the [MGLMapView class reference](https://www.mapbox.com/ios-sdk/api/3.7.3/Classes/MGLMapView.html#/MGLMapView) and seen in [this example](https://www.mapbox.com/ios-sdk/examples/point-conversion/) to be too onerous.